### PR TITLE
Feature/capture events after conn send

### PR DESCRIPTION
**Pull request by kagux**
This change is [Reviewable](https://reviewable.io/reviews/kagux/ex_debug_toolbar/9)

### CODE REVIEW

File: lib/ex_debug_toolbar/collector/instrumentation_collector.ex [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+  def ex_debug_toolbar_process_request_id(:start, _, _), do: :ok
+  def ex_debug_toolbar_process_request_id(:stop, _, _) do
+    Toolbar.start_request
+    Toolbar.start_event("request")
+  end
+
+  def ex_debug_toolbar(:start, _, _), do: :ok
+  def ex_debug_toolbar(:stop, _, _) do
+    Toolbar.finish_event("request")
```
Comment (epilgrim) [inline lib/ex_debug_toolbar/collector/instrumentation_collector.ex]: quick question before getting into the review. You are now wrapping the plug that generates the id.
Looks really asymmetrical. you do nothing in `ex_debug_toolbar(:start` nor in `ex_debug_toolbar_process_request_id(:start`

from an outsider point of view, the id should be generated in the `ex_debug_toolbar(:start`, and the request event started.
then, it should be finished in the `ex_debug_toolbar(:stop`
isn't that possible? that way, you don't need anymore the ex_debug_toolbar_process_request_id

File: test/lib/ex_debug_toolbar/phoenix_test.exs [hunk 2]
```diff
@@ -40,9 +22,15 @@ defmodule ExDebugToolbar.PhoenixTest do
     assert make_request().assigns[:called?] == true
   end
 
+  test "it tracks execution time of all following plugs in pipeline" do
+    make_request timeout: 100
+    assert {:ok, request} = get_request()
+    assert Timeline.duration(request.data.timeline) > 70 * 1000 # not sure why
```
Comment (epilgrim) [inline test/lib/ex_debug_toolbar/phoenix_test.exs]: wouldn't > 100 be enough?

File: test/lib/ex_debug_toolbar/plug/request_id_test.exs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  use ExUnit.Case, async: true
+  use Plug.Test
+  import Plug.Conn
+
+  test "it sets request id in response headers" do
+    [request_id] = build_conn() |> get_resp_header("x-request-id")
+    assert request_id
+  end
+
+  test "it sets request id in process metadata" do
```
Comment (epilgrim) [inline test/lib/ex_debug_toolbar/plug/request_id_test.exs]: this and the next one are exactly the same tests

Comment (kagux) [inline test/lib/ex_debug_toolbar/plug/request_id_test.exs]: this one is get_resp the next one is get_req
took me a while to notice it in request id plug

File: test/lib/ex_debug_toolbar/phoenix_test.exs (hunk 2 above)
Comment (kagux) [inline test/lib/ex_debug_toolbar/phoenix_test.exs]: duration is in microseconds, sleep timer is in milliseconds. 
but looking at test, duration/1 seems redundant now. 
what I couldn't figure out is why duration was almost always less than sleep time

**Pull request merged**